builds/fabric8-che/assembly/user-auth-valve/src/main/java/com/redhat/che/valve/UserAuthValve.java

```diff
@@ -41,15 +46,13 @@
  * @author amisevsk
  */
 public class UserAuthValve extends KeycloakAuthenticatorValve {
-
+    
     private static final Log LOG = LogFactory.getLog(UserAuthValve.class);
-    private static final String USER_VALIDATOR_ENDPOINT = "http://che-host:8080/api/token/user";
+    private static final String API_ENDPOINT = "http://che-host:8080/api";
+    private static final String USER_VALIDATOR_ENDPOINT = API_ENDPOINT + "/token/user";
     private static final String AUTHORIZATION_HEADER = "Authorization";
-    private static final String KEYCLOAK_SETTINGS_ENDPOINT = "http://che-host:8080/api/keycloak/settings";
-    private static final Pattern DISABLED_SETTING_PATTERN = Pattern.compile(".*\"che\\.keycloak\\.disabled\":\"([^ \"]+)\".*");
     
-    private boolean keycloakDisabledRetrieved = false;
-    private boolean keycloakDisabled = false;
+    private Boolean keycloakDisabled = null;
```

Comment:
yes I agree with you that the same value will be assigned but it looks odd to grab the settings several time. (so maybe it's not around Boolean/AtomicBoolean but how we may initialize the value several times.